### PR TITLE
fix falses while block hitting and switching slots

### PR DESCRIPTION
**Pull request by ManInMyVan**

### CODE REVIEW

File: src/main/java/ac/grim/grimac/events/packets/PacketPlayerAttack.java [hunk 1]
```diff
@@ -45,9 +47,11 @@ public void onPacketReceive(PacketReceiveEvent event) {
                 PacketEntity entity = player.compensatedEntities.getEntity(interact.getEntityId());
 
                 // You don't get a release use item with block hitting with a sword?
-                if (heldItem != null && player.getClientVersion().isOlderThan(ClientVersion.V_1_9)) {
-                    if (heldItem.getType().hasAttribute(ItemTypes.ItemAttribute.SWORD))
+                if (player.getClientVersion().isOlderThan(ClientVersion.V_1_9) && player.packetStateData.isSlowedByUsingItem()) {
+                    ItemStack item = player.getInventory().inventory.getPlayerInventoryItem(player.packetStateData.slowedByUsingItemSlot + HOTBAR_OFFSET);
```
Comment (SamB440) [inline src/main/java/ac/grim/grimac/events/packets/PacketPlayerAttack.java]: Are you certain this won't throw IndexOutOfBounds errors?

Comment (ManInMyVan) [reply src/main/java/ac/grim/grimac/events/packets/PacketPlayerAttack.java]: Shouldn't unless we change slowedByUsingItemSlot outside of the setter for slowedByUsingItem

File: src/main/java/ac/grim/grimac/utils/data/PacketStateData.java [hunk 2]
```diff
@@ -17,7 +17,9 @@ public class PacketStateData {
     public boolean tryingToRiptide = false;
     @Getter
     private boolean slowedByUsingItem;
-    public int slowedByUsingItemTransaction, slowedByUsingItemSlot = Integer.MIN_VALUE;
+    @Getter
+    private int slowedByUsingItemSlot = Integer.MIN_VALUE;
+    public int slowedByUsingItemTransaction = Integer.MIN_VALUE;
```
Comment (SamB440) [inline src/main/java/ac/grim/grimac/utils/data/PacketStateData.java:22]: Why is this min value? It started at 0 before

Comment (ManInMyVan) [reply src/main/java/ac/grim/grimac/utils/data/PacketStateData.java:22]: It was like that before the switch item patch

Comment (SamB440) [reply src/main/java/ac/grim/grimac/utils/data/PacketStateData.java:22]: It was 0 even at my cancel stupidity packets pr

Comment (ManInMyVan) [reply src/main/java/ac/grim/grimac/utils/data/PacketStateData.java:22]: Before #1445

**Review by SamB440 — APPROVED** [review]

**Pull request merged**